ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/ChainDB/Impl/ChainSel.hs

```diff
@@ -366,21 +366,22 @@ chainSelSync cdb@CDB {..} (ChainSelAddBlock BlockToAdd { blockToAdd = b, .. }) =
           blockPunish
           InvalidBlockPunishment.BlockItself
 
-        chainSelectionForFutureBlocks cdb BlockCache.empty
-
       -- The remaining cases
       | otherwise -> do
         let traceEv = AddedBlockToVolatileDB (blockRealPoint b) (blockNo b) isEBB
         lift $ encloseWith (traceEv >$< addBlockTracer) $
           VolatileDB.putBlock cdbVolatileDB b
         lift $ deliverWrittenToDisk True
+        -- REVIEW: would the tip returned by
```

Comment:
`getTipPoint` is just

```
getTipPoint CDB{..} =
    (castPoint . AF.headPoint) <$> readTVar cdbChain
```

So `castPoint $ AF.headPoint curChain` ought to suffice, in all of the branches of the `if |` that used to end with `chainSelectionForFutureBlocks` --- ChainSel is single threaded, so we know the tip didn't change by the end of those branches.

Reply:
Is the following correct then:

1. Setting the new tip as the value returned by `Query.getTipPoint cdb` **outside** the `if` branch should be equivalent to what we had before.
2. `chaniSelectionForBlock` should now return `  -> Electric m ()` (since we don't use it's return value anywhere with this change).